clients/src/test/java/org/apache/kafka/common/security/authenticator/SaslAuthenticatorFailureDelayTest.java

```diff
@@ -195,7 +195,7 @@ private TestJaasConfig configureMechanisms(String clientMechanism, List<String>
         return TestJaasConfig.createConfiguration(clientMechanism, serverMechanisms);
     }
 
-    private void createSelector(SecurityProtocol securityProtocol, Map<String, Object> clientConfigs) {
+    private void createSelector(SecurityProtocol securityProtocol, Map<String, Object> clientConfigs) throws IOException {
```

Comment:
I think quite a few of the files have changes that are related to either the propagation of IOException or the propagation of a watch service, both of which can be reverted.